nautobot/extras/jobs.py

```diff
@@ -979,21 +989,31 @@ def run_job(data, request, job_result_pk, commit=True, *args, **kwargs):
         job_result.completed = timezone.now()
         job_result.save()
         return False
-    job = job_class()
-    job.job_result = job_result
-
-    # Capture the file IDs for any FileProxy objects created so we can cleanup later.
-    file_fields = list(job._get_file_vars())
-    file_ids = [data[f] for f in file_fields]
 
-    # Attempt to resolve serialized data back into original form by creating querysets or model instances
-    # If we fail to find any objects, we consider this a job execution error, and fail.
-    # This might happen when a job sits on the queue for a while (i.e. scheduled) and data has changed
-    # or it might be bad input from an API request, or manual execution.
+    try:
+        job = job_class()
+        job.active_test = "initialization"
+        job.job_result = job_result
+    except Exception as e:
+        logger.error("Error initializing job object.")
+        logger.error(e)
+        job_result.set_status(JobResultStatusChoices.STATUS_ERRORED)
```

Comment:
Sorry for not catching this sooner - in this case, we have a job_result object, so just like the existing code at line 1018 below, we should log the exception to the job_result with `job_result.log_failure()` so that the user can see what went wrong without needing to refer to the Nautobot logs for context. Rather than logging just `str(e)`, I'd recommend something like the following to present a nicely formatted stack trace:

```suggestion
        stacktrace = traceback.format_exc()
        job_result.log_failure("Error initializing job:\n```\n{stacktrace}\n```")
        job_result.set_status(JobResultStatusChoices.STATUS_ERRORED)
```

Might want to update the log_failure call at line 1018 similarly as well.